refinery/core/views.py

```diff
@@ -832,8 +833,16 @@ def patch(self, request, uuid, format=None):
                 except Exception:
                     return Response(uuid, status=status.HTTP_404_NOT_FOUND)
 
-                self.data_set.transfer_ownership(current_owner, new_owner)
-                perm_groups = self.update_group_perms(new_owner)
+                try:
+                    with transaction.atomic():
+                        self.data_set.transfer_ownership(current_owner,
+                                                         new_owner)
+                        perm_groups = self.update_group_perms(new_owner)
+                except Exception as e:
+                    return Response(
+                        e, status=status.HTTP_500_INTERNAL_SERVER_ERROR
```

Comment:
Thinking again, this would be more appropriate as an error in the 400's. Its weird to explicitly send back a 500

Reply:
@scottx611x Suggestions?
http://www.django-rest-framework.org/api-guide/status-codes/#client-error-4xx

Reply:
It's not really a bad request is it?

Reply:
Maybe `HTTP_412_PRECONDITION_FAILED` ?

Reply:
Good reference for HTTP response status codes: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status